src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -420,6 +417,9 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
         }
     };  // compile
 
+    for (std::size_t i = 0u; i < orderedSubgraphs.size(); i++) {
+        m_compiled_submodels[i].device_it = m_dev_list.cbegin();
+    }
```

Comment:
Please don't do this, the current assumption is that if `compiled_submodels[i].compiled_model` is empty - there's no device associated with it - it should've been taken from the `replaced_by` one.

Reply:
Removed, thanks!